src/assets/wise5/components/table/edit-table-advanced/edit-table-advanced.component.ts

```diff
@@ -177,4 +183,91 @@ export class EditTableAdvancedComponent extends EditAdvancedComponentComponent {
       }
     }
   }
+
+  importTableFile(event: any): void {
+    if (confirm($localize`Are you sure you want to overwrite the existing table?`)) {
+      this.showImportingTableDisplay();
+      this.setImportTableMessage($localize`Importing table...`);
+      const files = event.target.files;
+      const reader: any = new FileReader();
+      reader.onload = () => {
+        const fileContent = reader.result;
+        const tableContent = this.UtilService.CSVToArray(fileContent);
+        const numCells = this.getNumCells(tableContent);
+        if (numCells > this.MAX_ALLOWED_CELLS_IN_IMPORT) {
+          this.setImportTableMessage(
+            $localize`Error: The table contains more than ${this.MAX_ALLOWED_CELLS_IN_IMPORT} cells`
+          );
+        } else {
+          this.importTable(tableContent);
+          this.setImportTableMessage($localize`Successfully imported table`);
+        }
+        event.target.value = null;
+        this.hideImportingTableDisplay();
+      };
+      reader.readAsText(files[0]);
+    } else {
+      event.target.value = null;
+    }
+  }
+
+  getNumCells(tableContent: string[][]): number {
+    let numCells = 0;
+    for (const row of tableContent) {
+      for (const cell of row) {
+        numCells++;
+      }
+    }
+    return numCells;
+  }
+
+  importTable(tableContent: string[][]): void {
+    const tableData = this.convertToTableData(tableContent);
+    this.authoringComponentContent.tableData = tableData;
+    this.authoringComponentContent.numRows = this.getNumRows(tableData);
+    this.authoringComponentContent.numColumns = this.getNumColumns(tableData);
+    this.componentChanged();
+  }
+
+  convertToTableData(stringArray: string[][]): any[][] {
+    const table = [];
+    for (const row of stringArray) {
+      const tableRow = [];
+      for (const cell of row) {
+        tableRow.push({
+          text: cell,
+          editable: true,
+          size: null
+        });
+      }
+      table.push(tableRow);
+    }
+    return table;
+  }
```

Comment:
```suggestion
  convertToTableData(stringArray: string[][]): any[][] {
    return stringArray.map((row) =>
      row.map((cell) => {
        return { text: cell, editable: true, size: null };
      })
    );
  }
```

Reply:
I left this as a double for loop since it will be more efficient when there are more cells. The double for loop is also less complex according to CodeMetrics.